### ISSUE TITLE
EOF while buffering

### ISSUE THREAD

**Issue by bluegaspode** [DigitalDJ/AudioStreamer#6]
Hi DigitalDJ, you seem to be the most active contributor to AudioStreamer, that's why I thouht I share my findings here.

I think there is one problem, when AudioStreamer is in the AS_BUFFERING state and then an EOF from the stream turnes up.
The streamer should be put in the AS_FLUSHING_EOF state then (at the same place when an EOF happens while the state is WAITING_FOR_DATA).

Otherwise the last bits of audio will never be streamed.

Can't provide a patch, as my own code did too many changes to the original already.

**Comment by jonnyijapan** [comment]
I agree. When playing from a file, the last seconds of a file will not be played, when EOF is hit. I'll have to try to find a fix for this... in the meantime I vote this up.
